### PR TITLE
Extract Jest config from package.json to config file 

### PR DESCRIPTION
**Pull request by thomasbertet**
Issue: https://github.com/storybooks/storybook/issues/2407

## What I did
 - Extract Jest config of _cra-kitchen-sink_ example  from _package.json_ to an external config file so that CRA is happy and let us run the test via `yarn run test`.
The idea is to have both **test** & **generate-addon-jest-testresults** scripts to work with no interactions/limitations.

## How to test
- Checkout all-the-things
- Go to _cra-kitchen-sink_ example project
- run `yarn run test` - it should run smoothly with some "expected" failure.
- then run `yarn run generate-addon-jest-testresults` it should generate the JSON results.

Is this testable with jest or storyshots?
Nope

Does this need a new example in the kitchen sink apps?
Nope

Does this need an update to the documentation?
Nope

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by thomasbertet** [comment]
ping @renaudtertrais can you PTAL and tell me what you think :thinking:

File: examples/cra-kitchen-sink/src/stories/addon-jest.testresults.json [hunk 1]
```diff
@@ -1 +0,0 @@
...umn":47},"end":{"line":94,"column":62}},"line":91},"11":{"name":"(anonymous_11)","decl":{"start":{"line":105,"column":48},"end":{"line":105,"column":49}},"loc":{"start":{"line":105,"column":60},"end":{"line":149,"column":1}},"line":105},"12":{"name":"(anonymous_12)","decl":{"start":{"line":143,"column":58},"end":{"line":143,"column":59}},"loc":{"start":{"line":143,"column":74},"end":{"line":143,"column":140}},"line":143},"13":{"name":"(anonymous_13)","decl":{"start":{"line":150,"column":55},"end":{"line":150,"column":56}},"loc":{"start":{"line":150,"column":67},"end":{"line":153,"column":62}},"line":150}},"branchMap":{"0":{"loc":{"start":{"line":4,"column":97},"end":{"line":4,"column":143}},"type":"cond-expr","locations":[{"start":{"line":4,"column":121},"end":{"line":4,"column":124}},{"start":{"line":4,"column":127},"end":{"line":4,"column":143}}],"line":4},"1":{"loc":{"start":{"line":4,"column":97},"end":{"line":4,"column":118}},"type":"binary-expr","locations":[{"start":{"line":4,"column":97},"end":{"line":4,"column":100}},{"start":{"line":4,"column":104},"end":{"line":4,"column":118}}],"line":4},"2":{"loc":{"start":{"line":19,"column":143},"end":{"line":19,"column":210}},"type"
```
Comment (Hypnosphi) [inline examples/cra-kitchen-sink/src/stories/addon-jest.testresults.json]: Why did you remove it?

Comment (thomasbertet) [reply examples/cra-kitchen-sink/src/stories/addon-jest.testresults.json]: How my! Will re-add it. no reason to remove it :+1:

**Review by Hypnosphi — APPROVED** [review]
LGTM, tested
